R/translate-sql-base.r

```diff
@@ -175,7 +175,42 @@ base_scalar <- sql_translator(
                         sql(ifelse(side == "both" | side == "right", "RTRIM(", "(")),
                         string
                         ,"))"
-                      )}
+                      )},
+
+  # lubridate functions
+
+  # SQL syntax reference links
+  # https://en.wikibooks.org/wiki/SQL_Dialects_Reference/Functions_and_expressions/Date_and_time_functions
+
+  # SQLite - https://www.sqlite.org/lang_datefunc.html
+  #        - https://www.sqlite.org/datatype3.html
+  #
+  as_date = sql_cast("DATE"),
+  as_datetime = sql_cast("TIMESTAMP"),
+
+  today = function(tzone = "") {build_sql("CURRENT_DATE")},
+  now = function(tzone = "") {build_sql("CURRENT_TIMESTAMP")},
+
+  year = function(x) {build_sql("EXTRACT(YEAR FROM ",x,")")},
+  month = function(x, label = FALSE, abbr = TRUE, locale = Sys.getlocale("LC_TIME")) {
+    # need to add label=TRUE AND abbr=TRUE, abbr=FALSE variants...
+    # i.e. Feb (3 chr), February
+    # to_char(current_date,'Mon'), to_char(current_date,'Month') ?
+    if (!label) {
+      return(build_sql("EXTRACT(MONTH FROM",x,")"))
+    } else {
+      if (abbr) {
+        return(build_sql("TO_CHAR(",x,",'Mon')"))
+      } else {
+        return(build_sql("TO_CHAR(",x,",'Month')"))
+      }
+    }
+
+  },
+  day = function(x){build_sql("EXTRACT(DAY FROM ",x,")")},
+  hour = function(x){build_sql("EXTRACT(HOUR FROM ",x,")")},
+  minute = function(x){build_sql("EXTRACT(MINUTE FROM ",x,")")},
+  second = function(x){build_sql("EXTRACT(SECOND FROM ",x,")")}
```

Comment:
Maybe it would be worth creating a custom function for these, especially since we'll probably use them again inside the translations for the other DB types, maybe something like this: `sql_prefix2("extract", "minute")`?

Reply:
I think `sql_expr(extract(day %from% !!x))` isn't too bad. If you were to create a wrapper function, I think I would recommend something specifically for `extract()`.